File: rudder-android-client/rudder-sdk-core/src/main/java/com/rudderlabs/android/sdk/core/event/RudderProperty.java
package com.rudderlabs.android.sdk.core.event;

import java.util.HashMap;
import java.util.Map;

public class RudderProperty {
    private Map<String, Object> map = new HashMap<>();

    public Object getProperty(String key) {
        return map.containsKey(key) ? map.get(key) : null;
    }

    public void setProperty(String key, Object value) {
        map.put(key, value);
    }

    public boolean hasProperty(String key) {
        return map.containsKey(key);
    }

    Map<String, Object> getMap() {
        return map;
    }

    public void setProperty(Map<String, Object> map) {
        this.map.putAll(map);
    }
}

File: rudder-client-android-kotlin-old/src/main/java/com/rudderlabs/android/library/models/porperties/RudderProperty.kt
package com.rudderlabs.android.library.models.porperties

import com.rudderlabs.android.library.util.convertToMap

open class RudderProperty {
    private val propertyMap = mutableMapOf<String, Any?>()

    fun getPropertyMap() = propertyMap

    fun containsKey(keyString: String): Boolean {
        return propertyMap.containsKey(keyString)
    }

    fun getProperty(key: String): Any? {
        return if (propertyMap.containsKey(key)) propertyMap[key] else null
    }

    fun addProperty(key: String, value: Any?) {
        propertyMap[key] = value
    }

    fun addProperties(map: Map<String, Any?>) {
        for (item in map) propertyMap[item.key] = item.value
    }

    fun <T> addListProperty(key: String) {
        if (!propertyMap.containsKey(key)) {
            propertyMap[key] = mutableListOf<T?>()
        }
    }

    fun <T> addListItem(key: String, item: T) {
        if (propertyMap.containsKey(key)) {
            (propertyMap[key] as MutableList<T>).add(item)
        }
    }

    fun <T> addListItems(key: String, items: List<T>) {
        if (propertyMap.containsKey(key)) {
            (propertyMap[key] as MutableList<T>).addAll(items)
        }
    }

    fun <T> addAsMap(item: T) {
        addProperties(item.convertToMap())
    }
}